src/Event/Decorator/BaseDecorator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * CakePHP : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Event\Decorator;
+
+class BaseDecorator
```

Comment:
> Usually base classes are abstract since they are don't have much value by itself. 

Agreed and already done.

> Makes sense or do you see other decorators not using this method?

Honestly, I have no idea. :) This is something time will show.